src/bidiMapper/domains/context/BrowsingContextImpl.ts

```diff
@@ -704,52 +707,105 @@ export class BrowsingContextImpl {
     await this.#cdpTarget.cdpClient.sendCommand('Page.bringToFront');
   }
 
-  async captureScreenshot(): Promise<BrowsingContext.CaptureScreenshotResult> {
+  async captureScreenshot(
+    params: BrowsingContext.CaptureScreenshotParameters
+  ): Promise<BrowsingContext.CaptureScreenshotResult> {
+    if (!this.isTopLevelContext()) {
+      throw new UnsupportedOperationException(
+        `Non-top-level 'context' is currently not supported`
+      );
+    }
+
     // XXX: Focus the original tab after the screenshot is taken.
     // This is needed because the screenshot gets blocked until the active tab gets focus.
     await this.#cdpTarget.cdpClient.sendCommand('Page.bringToFront');
 
-    let clip: Protocol.DOM.Rect;
-
-    if (this.isTopLevelContext()) {
-      const {cssContentSize, cssLayoutViewport} =
-        await this.#cdpTarget.cdpClient.sendCommand('Page.getLayoutMetrics');
-      clip = {
-        x: cssContentSize.x,
-        y: cssContentSize.y,
-        width: cssLayoutViewport.clientWidth,
-        height: cssLayoutViewport.clientHeight,
-      };
-    } else {
-      const {
-        result: {value: iframeDocRect},
-      } = await this.#cdpTarget.cdpClient.sendCommand(
-        'Runtime.callFunctionOn',
-        {
-          functionDeclaration: String(() => {
-            const docRect =
-              globalThis.document.documentElement.getBoundingClientRect();
-            return JSON.stringify({
-              x: docRect.x,
-              y: docRect.y,
-              width: docRect.width,
-              height: docRect.height,
-            });
-          }),
-          executionContextId: this.#defaultRealm.executionContextId,
+    const sandbox = await this.getOrCreateSandbox(undefined);
```

Comment:
Note: I was thinking about this and we should create a `BiDi Mapper` world and evaluate things there as user can remove things on the main one.
No need to action on this but please put a comment.

Reply:
Indeed, we should be doing this. I'll create an issue; there are other places where this occurs but don't have a comment. I think we can manage in one swoop later by looking at the issue (rather than some comment).